cpp/src/io/parquet/reader_impl.hpp

```diff
@@ -94,7 +94,7 @@ class reader::impl {
    * @return The total number of pages
    */
   size_t count_page_headers(
-      const hostdevice_vector<gpu::ColumnChunkDesc> &chunks,
+      hostdevice_vector<gpu::ColumnChunkDesc> &chunks,
```

Comment:
Why are a bunch of `const&` being changed to non-const refs?

Reply:
Because the old implementation of hostdevice_vector returned a non-const pointer to device memory even for const objects. I modified it to return a const pointer (as I think it should). This broke many call sites, where a const ref was passed even though the memory owned by the object would be changed in the function.

Reply:
Oh, yikes, yeah, your change sounds correct.